base/abstractset.jl

```diff
@@ -119,7 +119,7 @@ Set{Int64} with 1 element:
 """
 intersect(s::AbstractSet, itr, itrs...) = intersect!(intersect(s, itr), itrs...)
 intersect(s) = union(s)
-intersect(s::AbstractSet, itr) = mapfilter(_in(s), push!, itr, emptymutable(s))
+intersect(s::AbstractSet, itr) = length(s) > length(itr) ? mapfilter(_in(s), push!, itr, emptymutable(s)) : mapfilter(_in(itr), push!, s, emptymutable(itr))
```

Comment:
Thanks for the patch!  Unfortunately, this has a few problems:

* `itr` may not define `length`. We need to support generic iterables.  You can use `haslength(itr)` to check if you can call `length(itr)`.
* `emptymutable(itr)` may not return a set. We should always use `emptymutable(s)`.

Reply:
Also, we need to check `hasfastin(itr)` if we can use `_in(itr)`.

Finally, please limit the length of a line to 92 characters (as mentioned in CONTRIBUTING.md)

Reply:
does it look better now? (force pushed